flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamPhysicalDeduplicate.scala

```diff
@@ -30,9 +30,10 @@ import org.apache.calcite.rel.{RelNode, RelWriter, SingleRel}
 import java.util
 
 /**
- * Stream physical RelNode which deduplicate on keys and keeps only first row or last row. This node
- * is an optimization of [[StreamPhysicalRank]] for some special cases. Compared to
- * [[StreamPhysicalRank]], this node could use mini-batch and access less state.
+ * TODO to be removed after FLINK-34702 is fixed. Stream physical RelNode which deduplicate on keys
```

Comment:
Since this operator may be used in existing Flink jobs, I do not believe we can remove it.

Reply:
The `StreamPhysicalDeduplicate` is just an intermediate node in the optimization process, after all the optimizations are done it will be converted to `StreamExecDeduplicate` and eventually to the actual runtime sql duplicate operator.
Also, the compiled plan does not depend on the physical node, but on the exec node. Therefore, it does not affect existing sql jobs.